src/js/components/TabPanesComponent.jsx

```diff
@@ -33,15 +47,23 @@ var TabPanesComponent = React.createClass({
 
     return (
       <TogglableTabsComponent activeTabId={this.getTabId()}
-        className="container-fluid">
+                              className="container-fluid">
         <TabPaneComponent id={tabs[0].id}>
-          <Link
+          <div className="row">
+            <div className="col-md-3">
+              <AppListFilterComponent onChange={this.updateFilterText}/>
+            </div>
+            <div className="col-md-4 col-md-offset-5">
+              <Link
               to={path}
               query={{modal: "newapp"}}
-              className="btn btn-success navbar-btn">
-            + New App
-          </Link>
-          <AppListComponent />
+              className="btn btn-success"
```

Comment:
I'm guessing we lost the `navbar-btn` class here so that the filter component and the button would be vertically aligned. It would be good to have the 8px spacing at the top and bottom which this was providing re-introduced - a wrapping `div` or similar with 8px of vertical padding would do the job nicely.